storage/src/storage/sync.rs

```diff
@@ -246,40 +246,16 @@ pub trait SyncStorage {
         }
 
         // Calculate the average distance between block hashes based on the desired number of locator hashes.
-        let mut proportional_step =
+        let proportional_step =
             (hash_index.min(oldest_fork_threshold as u32) / num_locator_hashes).min(crate::NUM_LOCATOR_HASHES - 1);
 
         // Provide hashes of blocks with indices descending quadratically while the quadratic step distance is
         // lower or close to the proportional step distance.
-        let num_quadratic_steps = (proportional_step as f32).log2() as u32;
+        let num_quadratic_steps = ((hash_index as f32).log2() - (proportional_step as f32).log2()) as u32;
```

Comment:
I'd update the comment to document this change. 

(The log subtraction could also be shorter if replaced by a division inside the log, though this may be less clear, just mentioning the option.)